crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -749,12 +760,20 @@ async fn process_smartrest(
     operations: &Operations,
     http_proxy: &mut impl C8YHttpProxy,
     operation_logs: &OperationLogs,
+    device_name: &str,
 ) -> Result<Vec<Message>, CumulocityMapperError> {
-    let message_id = get_smartrest_template_id(payload);
-    match message_id.as_str() {
-        "528" => forward_software_request(payload, http_proxy).await,
-        "510" => forward_restart_request(payload),
-        template => forward_operation_request(payload, template, operations, operation_logs).await,
+    match get_smartrest_device_id(payload) {
+        Some(device_id) if device_id == device_name => {
+            match get_smartrest_template_id(payload).as_str() {
+                "528" => forward_software_request(payload, http_proxy).await,
+                "510" => forward_restart_request(payload),
+                template => {
+                    forward_operation_request(payload, template, operations, operation_logs).await
+                }
+            }
+        }
+        // Ignore all operations for child devices as not yet supported
+        _ => Ok(vec![]),
```

Comment:
You could write a unit-test for this child device case where you expect an empty vector. Either for this function itself or for the `try_convert` function which calls this one.

Reply:
Committed.